pkg/knit/cmd/k8s/podinfo.go

```diff
@@ -0,0 +1,204 @@
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ */
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/user"
+	"path/filepath"
+	"text/template"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/openshift-kni/debug-tools/pkg/knit/cmd"
+	"github.com/spf13/cobra"
+)
+
+type podInfoOptions struct {
+	nodeName string
+}
+
+//Only need some info about the pod.
+// Right now is:
+// - pod name
+// - pod namespace
+// - node name
+// - status.qosClass
+// - containers
+//   - requests cpu
+//   - limits cpu
+// Note this output format could change but it would be parsed on insight rules
+// so the change should be sync with it.
+// Caution: We filter the data from pods to avoid exposing sensible information
+// (like environment variables or input parameters which can contain passwords)
+// so take care of that when changing this template.
+const default_template string = `
+[
+	{{- range $idx, $item := .Items}}
+	{{- if (ne $idx 0)}},{{end}}
+	{
+		"namespace":"{{.ObjectMeta.Namespace}}", 
+		"name":"{{.ObjectMeta.Name}}", 
+		"nodeName":"{{.Spec.NodeName}}", 
+		"qosClass": "{{.Status.QOSClass}}", 
+		{{- if .Spec.Containers }}
+		"containers": [
+		{{- range $cdx, $cont := .Spec.Containers -}}
+			{{- if (ne $cdx 0) }},{{ end }} 
+			{
+				"name":"{{.Name}}"
+				{{- if or .Resources.Requests .Resources.Limits -}}
+				,
+				"resources": {
+					{{- if .Resources.Limits}} {{if .Resources.Limits.Cpu}} 
+					"limits": {
+						"cpu": "{{.Resources.Limits.Cpu}}"
+					}
+					{{- end }}{{end}}
+					{{- if .Resources.Requests}}{{if .Resources.Requests.Cpu -}}
+					,
+					"requests": {
+						"cpu": "{{.Resources.Requests.Cpu}}"
+					}
+					{{- end }}{{end}}
+				} 
+				{{- end }} 
+			} 
+		{{- end }} 
+		]
+		{{- end }} 
+	}
+	{{- end }}
+]`
+
+func NewPodInfoCommand(knitOpts *cmd.KnitOptions) *cobra.Command {
+
+	opts := &podInfoOptions{}
+	podInfo := &cobra.Command{
+		Use:   "podinfo",
+		Short: "get pod status/spec information from cluster",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			clientset, err := getClientSetFromClusterConfig()
+			if err != nil {
+				return fmt.Errorf("unable to get clientset: %w", err)
+			}
+
+			podInfoTemplate, err := createOutputTemplate("pod_info", default_template)
+			if err != nil {
+				return fmt.Errorf("unable to get output template: %w", err)
+			}
+
+			nodeFieldSelector := buildNodeFieldSelector(opts.nodeName)
+
+			return showPodInfo(nodeFieldSelector, clientset, podInfoTemplate, os.Stdout)
+		},
+	}
+
+	podInfo.Flags().StringVar(&opts.nodeName, "node-name", "", "node name to get pod info from.")
+
+	return podInfo
+}
+
+// GetConfig creates a *rest.Config for talking to a Kubernetes apiserver.
+//
+// Config precedence
+//
+// - KUBECONFIG environment variable pointing at a file
+// - In-cluster config if running in cluster
```

Comment:
why in-cluster is between the (presumably) outside cluster options?

Reply:
I am having some problems loading the configuration.

My first try has been to use "sigs.k8s.io/controller-runtime/pkg/client/config" `config.GetConfig()` as in NTO and the second one has been to use this function as it is used in PAO but I keep getting the same error while building and I do not know what the problem is....

```
14:51 $ make all
mkdir -p _output || :
go mod tidy && go mod vendor
# go flags are set in here
./hack/build-binaries.sh
go build -v -o "/home/jojosneg/sources/redhat/root/debug-tools/main/hack/../_output/cpulist" "/home/jojosneg/sources/redhat/root/debug-tools/main/hack/../cmd/cpulist"                                                                                                                   
github.com/openshift-kni/debug-tools/cmd/cpulist
go build -v -o "/home/jojosneg/sources/redhat/root/debug-tools/main/hack/../_output/knit" "/home/jojosneg/sources/redhat/root/debug-tools/main/hack/../cmd/knit"                                                                                                                         
k8s.io/client-go/applyconfigurations/meta/v1
# k8s.io/client-go/applyconfigurations/meta/v1
vendor/k8s.io/client-go/applyconfigurations/meta/v1/unstructured.go:64:38: cannot use doc (variable of type *"github.com/googleapis/gnostic/openapiv2".Document) as type *"github.com/google/gnostic/openapiv2".Document in argument to proto.NewOpenAPIData                             
make: *** [Makefile:34: binaries] Error 2
```

Reply:
> why in-cluster is between the (presumably) outside cluster options?

Just used the function as It is in PAO ... but maybe I should include this function in a common point to use the same code from different projects instead of repeating it. 
what do you think? 
is there any obvious candidate?

Reply:
I mean it is ok with me. In fact your way seems better to me, I just used that order because I thought it was done that way for a reason.